HW21/TestRunner.java

```diff
@@ -0,0 +1,53 @@
+package HW21;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class TestRunner {
+
+    public static void start(Class<?> classe) throws Exception {
+        Object instance = classe.getDeclaredConstructor().newInstance();
+
+        Method[] methods = classe.getDeclaredMethods();
+        List<Method> beforeSuiteMethods = new ArrayList<>();
+        List<Method> testMethods = new ArrayList<>();
+        List<Method> afterSuiteMethods = new ArrayList<>();
+
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(BeforeSuite.class)) {
+                beforeSuiteMethods.add(method);
+            } else if (method.isAnnotationPresent(Test.class)) {
+                testMethods.add(method);
+            } else if (method.isAnnotationPresent(AfterSuite.class)) {
+                afterSuiteMethods.add(method);
+            }
+        }
+
+        if (beforeSuiteMethods.size() > 1) {
+            throw new RuntimeException("BeforeSuite method should be declared only once in a class");
+        }
+
+        if (afterSuiteMethods.size() > 1) {
+            throw new RuntimeException("AfterSuite method should be declared only once in a class");
+        }
+
+        beforeSuiteMethods.sort(Comparator.comparingInt(m -> m.getAnnotation(BeforeSuite.class).priority()));
```

Comment:
If you started to work with the collection in a "functional way", - do it till the end. For example, here you sort "before suit" methods and a bit later execute them in for loop - why you can do sorting and execution in one pipe? Something like: 
```
beforeSuiteMethods
  .sort(Comparator.comparingInt(m -> m.getAnnotation(BeforeSuite.class).priority())) 
  .forEach(method -> method.invoke(instance));
```